tcms/testplans/tests/test_admin.py

```diff
@@ -40,3 +40,37 @@ def test_users_can_delete_testplan(self):
         )
         self.assertRedirects(response, reverse("core-views-index"))
         self.assertEqual(TestPlan.objects.filter(pk=self.test_plan.pk).exists(), False)
+
+    def test_history_display_before_change_testplan(self):
+        response = self.client.get(
+            "/admin/testplans/testplan/%d/history/" % self.test_plan.pk
+        )
+
+        self.assertContains(response, "<pre></pre>")
+        self.assertContains(response, _("<td>Created</td>"))
+
+    def test_history_display_and_redirect_after_change_testplan(self):
+        old_test_plan_name = self.test_plan.name
+        with self.subTest("Test change done successfully"):
+            self.test_plan.name = "Test Demo"
+            self.test_plan.save()
+            self.assertEqual(self.test_plan.name, "Test Demo")
+
+        with self.subTest("Test changes included in history"):
+            response = self.client.get(
+                "/admin/testplans/testplan/%d/history/" % self.test_plan.pk
+            )
+            self.assertContains(
+                response,
+                f"<pre>--- name\n+++ name\n@@ -1 +1 @@\n-{old_test_plan_name}\n+Test Demo</pre>",
+            )
+
+        with self.subTest("Test Reverting Redirect to Object"):
+            response = self.client.post(
+                "/admin/testplans/testplan/%d/history/1/" % self.test_plan.pk,
+                follow=True,
+            )
+            self.assertRedirects(
+                response,
+                reverse("test_plan_url", args=[self.test_plan.pk, old_test_plan_name]),
```

Comment:
The second argument is not the name verbatim but a slug. The URL regex imposes limitation to the available characters that's why the assertion fails.

See also `self.test_plan._get_absolute_url()`